README.md

```diff
@@ -194,60 +194,106 @@ populate the following:
 
+## Using the development environment
+
+The Docker Compose stack we use to run Spark (via `docker compose up -d`)
+has a separate development environment that can be used to test code changes
+without disrupting the production containers.
+
+The development environment uses
+[Compose profiles](https://docs.docker.com/compose/how-tos/profiles/) to run
+a parallel set of containers with tweaked setup/environment variables.
+
+To start the development environment, run:
+
+```bash
+docker compose --profile dev up -d
+```
+
+To submit a job to the development environment, change the container target
+of your command from `prod` to `dev`. For example:
+
+```bash
+docker exec -it spark-node-master-dev ./submit.sh \
+    --json-string "$(yq -o=json .test_jobs ./config/default_jobs.yaml)"
+```
+
+A typical development workflow might look something like:
+
+1. Clone the repository to your own machine or home directory. Do _not_ use
+  the production `shiny-server` copy of the repository for development.
+2. Copy the secrets, environmental variables, and drivers from the production
+  setup to the development repository. See [Files not included](#files-not-included).
+3. Build the Docker container targeting the `dev` tag by running
+  `docker compose --profile dev build`.
+4. Start the development environment using `docker compose --profile dev up -d`.
+5. Make your code modifications. Changes in the `src/` directory are reflected
+  in the dev containers due to volume mounts (no need to rebuild).
+6. Submit a job to the development containers using `docker exec`, targeting
+  the development master node (`spark-node-master-dev`).
+7. Check the job status at `$SERVER_IP:8082`, instead of the production port
+  `$SERVER_IP:8080`.
+
+> [!WARNING]
+> The development environment shares the same targets as the production
+> environment. That means it will write to the same S3 bucket/CloudWatch log
+> group and trigger the same workflows/crawlers (though all these features are
+> disabled by default). As such, use this environment carefully. If you mess up
+> production data, you can run the production version of the code to re-fetch it.
+
 ## Scheduling
 
 Batches are currently scheduled via
 [`cron`](https://man7.org/linux/man-pages/man8/cron.8.html). To edit the
 schedule file, use `crontab -e` as the main server user. The example crontab
 file below schedules daily jobs for frequently updated tables and weekly ones
-for rarely-updated tables. Note that the jobs currently _must_ be run as
-user 1003.
+for rarely-updated tables.
 
 ```bash
 # Extract recent years from frequently used tables on weekdays at 1 AM CST
-0 6 * * 1,2,3,4,5 docker exec spark-node-master ./submit.sh --json-string "$(yq -o=json .default_jobs /full/path/to/default_jobs.yaml)"
+0 6 * * 1,2,3,4,5 docker exec spark-node-master-prod ./submit.sh --json-string "$(yq -o=json .default_jobs /full/path/to/default_jobs.yaml)" --run-github-workflow --run-glue-crawler --upload-data --upload-logs
```

Comment:
Note that we need to explicitly set all these flags since they're now `False` by default. We'll also need to update the actual cron jobs once this is merged.